app/wasm/message_plugin.go

```diff
@@ -165,10 +165,18 @@ func parseAddress(addr string) (sdk.AccAddress, error) {
 	return parsed, nil
 }
 
+const reSubdenomStr = `^[a-zA-Z][a-zA-Z0-9]{2,31}$`
+
+var reSubdenom *regexp.Regexp
+
+func init() {
```

Comment:
This being called automatically as part of initialisation, nice.

Reply:
Yup, one time overhead to compile the regexp, not each call.
This seems a safe use of globals, which I usually avoid.